crates/core/tedge_monitor/src/main.rs

```diff
@@ -0,0 +1,150 @@
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+use mqtt_channel::{Config, Message, MqttError, PubChannel, Topic};
+use nanoid::nanoid;
+use std::process::{self, Command, Stdio};
+
+use serde::{Deserialize, Serialize};
+use tedge_config::{
+    ConfigRepository, ConfigSettingAccessor, ConfigSettingError, MqttBindAddressSetting,
+    MqttPortSetting, TEdgeConfigError,
+};
+use tokio::time::Duration;
+
+pub const DEFAULT_TEDGE_CONFIG_PATH: &str = "/etc/tedge";
+
+#[derive(Serialize, Deserialize)]
+pub struct Response {
+    status: String,
+    pid: u32,
+}
+#[tokio::main]
+async fn main() {
+    let _ = start_watchdog().await;
+}
+
+async fn start_watchdog() {
+    // Send ready notification to systemd.
+    let _ = notify_systemd(process::id(), "--ready");
+
+    // Start helth check request publisher
+    tokio::spawn(async move { publish().await });
```

Comment:
A dedicated task just doing the publishes at regular intervals and the response processing in another independent task, huh? Smart. Not quite the way I had imagined. I was expecting both to happen in a single loop. But I don't think this approach has any problems either.